### PR TITLE
Add discussion bullet points to README

### PR DESCRIPTION
**Pull request by jerrymarino**
The README has been pretty out of date with with this repo actually does and is aiming to do

Addresses how people use it for Xcode generatiion and other abilities https://github.com/bazel-ios/rules_ios/discussions/564

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,31 +2,107 @@
...
-allowing users to customize things such as generated modulemaps.
+`rules_ios` is community developed Bazel rules to do iOS development with Bazel
+end to end.
 
-Bazel version required by current rules is [here](https://github.com/bazel-ios/rules_ios/blob/master/.bazelversion)
+It seamlessly bazel builds iOS applications originally written under Xcode with
+minimal-to-no code changes. It often re-uses ideas and code from `rules_swift`
+and `rules_apple` buit it isn't tied to untested or unused features. It
+generates Xcode projects that _just work_ and makes using Apple Silicion a
+with Bazel breeze.
```
Comment (luispadron) [inline README.md:12]:
```suggestion
generates Xcode projects that _just work_ and makes using Apple Silicion with Bazel a breeze.
```

**Review by luispadron — APPROVED** [review]
Nice!

File: README.md [hunk 2]
```diff
@@ -2,31 +2,102 @@
...
+    srcs = glob(["*.m", "*.swift"]),
+    bundle_id = "com.example.ios-app",
+    minimum_os_version = "12.0",
+    visibility = ["//visibility:public"],
+)
+```
+
+### Xcode project generation
+
+_xcode project generaation that's tested and works end to end with open source
```
Comment (thiagohmcruz) [inline README.md:31]: `generaation` => `generation`.

I think mentioning remote execution here is not necessary and might confuse new folks but that's just my opinion, feel free to keep it and treat as a nit...

File: README.md [hunk 3]
```diff
@@ -2,31 +2,102 @@
...
+remote execution and caching._
 
-## Quick setup
+```python
+load("@build_bazel_rules_ios//rules:xcodeproj.bzl", "xcodeproj")
+
+xcodeproj(
+    name = "MyXcode",
+    srcs = glob(["*.m", "*.swift"]),
```
Comment (thiagohmcruz) [inline README.md:39]: IIRC `srcs` is not a valid arg in this rule?

File: README.md [hunk 4]
```diff
@@ -2,31 +2,102 @@
...
+
+```python
+# Builds a static framework
+apple_framework(
+    name = "Static",
+    srcs = glob(["static/*.swift"]),
+    bundle_id = "com.example.b",
+    data = ["Static.txt"],
+    infoplists = ["Info.plist"],
+    link_dynamic = True,
```
Comment (thiagohmcruz) [inline README.md:57]: Should `link_dynamic` be `True` here?

**Review by thiagohmcruz — APPROVED** [review]
Minor comments inline. Nit: `easily` is repeated too many times.

I'd double check all examples to make sure they actually work before landing.

**Comment by thiagohmcruz** [comment]
Also, should we link the website in the `README`?

https://bazel-ios.github.io/

File: README.md (hunk 3 above)
Comment (jerrymarino) [reply README.md]: Good catch - fixed 🏁

File: README.md (hunk 2 above)
Comment (jerrymarino) [reply README.md]: The reason I mention remote caching and execution - is that the rules and projects have specific changes to actually work with these

**Comment by jerrymarino** [comment]
> Minor comments inline. Nit: easily is repeated too many times.

Good suggestion, let me clean some up 👍 

> Also, should we link the website in the README?

Yep love that idea - social proofing these rules / Bazel is a direct goal!

File: README.md [hunk 5]
```diff
@@ -2,38 +2,112 @@
...
+
+## WORKSPACE setup
 
 Add the following lines to your `WORKSPACE` file. Note that since `rules_swift`
 and `rules_apple` [no longer create
 releases](https://github.com/bazelbuild/rules_swift/pull/335), the versions are
 hardcoded to commit sha's that are known to work. You can see the particular
 commit sha's in
 [`repositories.bzl`](https://github.com/bazel-ios/rules_ios/tree/master/rules/repositories.bzl).
+_It pulls a vetted sha of `rules_apple` and `rules_swift`._
```
Comment (mattrobmattrob) [inline README.md:110]: In general, this line seems out of date now. For example:

- https://github.com/bazelbuild/rules_swift/releases/tag/1.2.0
- https://github.com/bazelbuild/rules_apple/releases/tag/1.1.2

**Review by mattrobmattrob — APPROVED** [review]
This is a big improvement. 👏

File: README.md (hunk 5 above)
Comment (thiagohmcruz) [reply README.md:110]: +1 this seems out of date indeed

Comment (jerrymarino) [reply README.md:110]: Good point  - let me bump this as well!

File: README.md (hunk 4 above)
Comment (jerrymarino) [reply README.md:59]: Fixed

**Pull request merged**